src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java

```diff
@@ -184,14 +201,35 @@ private Mono<TransactionActivatedEvent> newTransactionActivatedEvent(
     data.setEmail(email);
     data.setPaymentToken(paymentToken);
 
-    TransactionActivatedEvent transactionActivationRequestedEvent =
+    TransactionActivatedEvent transactionActivatedEvent =
         new TransactionActivatedEvent(transactionId, rptId, paymentToken, data);
 
     log.info(
         "Generated event TRANSACTION_ACTIVATED_EVENT for rptId {} and transactionId {}",
         rptId,
         transactionId);
 
-    return transactionEventActivatedStoreRepository.save(transactionActivationRequestedEvent);
+    return transactionEventActivatedStoreRepository
+        .save(transactionActivatedEvent)
+        .thenReturn(transactionActivatedEvent)
+        .doOnNext(
+            event ->
+                transactionActivatedQueueAsyncClient
+                    .sendMessageWithResponse(
+                        BinaryData.fromObject(event),
+                        Duration.ofSeconds(Integer.valueOf(paymentTokenTimeout)),
+                        null)
+                    .subscribe(
+                        response ->
+                            log.debug(
+                                "TransactionActivatedEvent {} expires at {} for transactionId {}",
+                                response.getValue().getMessageId(),
+                                response.getValue().getExpirationTime(),
+                                transactionActivatedEvent.getTransactionId()),
+                        error -> log.error(error.toString()),
+                        () ->
+                            log.debug(
+                                "Complete enqueuing the message TransactionActivatedEvent for transactionId {}!",
+                                transactionActivatedEvent.getTransactionId())));
```

Comment:
I'm worried that `doOnNext` may break error chaining from the `sendMessageWithResponse` Mono to the outer Mono, maybe `onErrorResume` may be a better choice of combinator

Reply:
this event is sent if the saving on the eventstore is successful, and then the event on the queue is sent (from this event a "virtual" timer starts to evaluate the pending transaction). I understand your point, so better `doOnSuccess`?